### PR TITLE
Use `rapidsai/miniforge-cuda` as image base

### PR DESCRIPTION
**Pull request by charlesbluca**

### CODE REVIEW

**Review by pentschev — COMMENTED** [review]
If this is not of urgent need I'd suggest we wait before making any changes to Dask gpuCI. The reason for that is we were still pinned to `aws-sdk-cpp` version that was not causing segfaults for us. Now what's in the pipeline is to upgrade cuDF to [Arrow 13](https://github.com/conda-forge/arrow-cpp-feedstock/pull/1211) that has the patch we need to stop initialization of S3 on import and thus resolve the segfault issues.

**Comment by pentschev** [comment]
xref the cuDF PR upgrading to Arrow 13: https://github.com/rapidsai/cudf/pull/14330.

**Comment by charlesbluca** [comment]
Yeah this isn't urgent and I'm fine to hold off on merging

File: dask/Dockerfile [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
 COPY environment.yml /rapids.yml
 ADD https://raw.githubusercontent.com/dask/dask/main/continuous_integration/environment-$PYTHON_VER.yaml /dask.yml
 
 RUN conda config --set ssl_verify false
 
-RUN conda install -c gpuci gpuci-tools
+RUN mamba install -c gpuci gpuci-tools
 
-RUN gpuci_conda_retry install -c conda-forge mamba conda-merge "conda>=4.14"
+RUN gpuci_mamba_retry install conda-merge git
```
Comment (ajschmidt8) [inline dask/Dockerfile:19]: anything related to `gpuci` has been deprecated.

We have some similar tools below that we use for the RAPIDS CI images that you could install if you want:

- https://github.com/rapidsai/gha-tools

The `gpuci_*` commands have all been superseded by `rapids-*` commands.

File: dask_image/Dockerfile [hunk 2]
```diff
@@ -30,5 +31,4 @@ RUN chmod -R ugo+w /opt/conda \
     && conda clean -tipy \
     && chmod -R ugo+w /opt/conda
```
Comment (ajschmidt8) [inline dask_image/Dockerfile:32]: once `dask` is no longer using Jenkins, all of these `chmod` commands can be removed across the Dockerfiles.

`chmod` artificially inflates the size of Docker images: https://medium.com/@lmakarov/the-backlash-of-chmod-chown-mv-in-your-dockerfile-f12fe08c0b55

**Comment by pentschev** [comment]
@charlesbluca Dask-CUDA is now unblocked, so feel free to move forward with this change when convenient, thanks for holding this!

**Review by quasiben — APPROVED** [review]

**Pull request merged**